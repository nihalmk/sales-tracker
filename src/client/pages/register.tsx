import React from 'react';
import { NextPage } from 'next';
import { Layout } from '../components/Layout/Layout';
import Register from '../components/User/Register';

interface Props {}

const RegisterPage: NextPage<Props> = () => {

  return (
    <React.Fragment>
      <Layout hideHeader={false}><Register /></Layout>
    </React.Fragment>
  );
};

export default RegisterPage;
