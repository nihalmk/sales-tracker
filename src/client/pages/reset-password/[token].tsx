import React, { useState } from 'react';
import { NextPage } from 'next';
import { Logo } from '../../components/Header/Logo';
import { accountsPassword } from '../../../accounts/client';
import { useRouter } from 'next/router';
import cookie from 'js-cookie';

interface Props {}

export const Reset: NextPage<Props> = () => {
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [isLoading, setLoading] = useState(false);
  const router = useRouter();
  const token: string = router.query.token.toString();

  const onPasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setPassword(event.target.value);
  };
  const onConfirmPasswordChange = (
    event: React.ChangeEvent<HTMLInputElement>,
  ) => {
    setConfirmPassword(event.target.value);
  };

  const changePassword = async () => {
    setLoading(true);
    try {
      const resp = await accountsPassword.resetPassword(token, password);
      cookie.remove('token');
      cookie.set('token', resp.tokens.accessToken, { expires: 1 });
      router.push('/');
    } catch (ex) {
      router.push('/forgot?from=reset');
    }
  };

  return (
    <div className="col col-login mx-auto">
      <div className="text-center mb-6">
        <Logo setColor />
      </div>
      <div className="card">
        <div className="card-body">
          <div className="card-title text-center">Reset Password</div>
        </div>
        <div className="card-body p-6">
          <p>
            Enter your new password below to reset the password of your account.
          </p>
          <div className="form-group">
            <label className="form-label">New Password</label>
            <input
              name="password"
              type="password"
              className="form-control"
              id="password"
              placeholder="Password"
              onChange={onPasswordChange}
            />
          </div>
          <div className="form-group">
            <label className="form-label">Confirm New Password</label>
            <input
              name="confirmPassword"
              type="password"
              className="form-control"
              id="confirmPassword"
              placeholder="Password"
              onChange={onConfirmPasswordChange}
            />
          </div>
          <div className="form-footer">
            <button
              type="button"
              onClick={changePassword}
              className={
                'btn btn-primary btn-block ' + (isLoading ? 'btn-loading' : '')
              }
              disabled={password === '' || password !== confirmPassword}
            >
              Reset password
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Reset;
