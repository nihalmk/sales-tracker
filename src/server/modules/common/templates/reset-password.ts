import { User } from '@accounts/types';

export interface EmailTemplateType {
  subject: (user?: User | null) => string;
  text: (user: User | null, url: string) => string;
  html: (user: User | null, url: string) => string;
}

export const ResetPasswordTemplate: EmailTemplateType = {
  subject: () => `Password reset request`,
  text: (_user: User | null, url: string) => {
    return `To reset your password, please click on this link: ${url.replace(
      'reset-password',
      'reset',
    )}`;
  },
  html: (_user: User | null, url: string) => {
    return `<div style="margin:0;padding:0;background:#f5f7fb">
    <table
      width="100%"
      height="100%"
      style="min-width:348px"
      border="0"
      cellspacing="0"
      cellpadding="0"
      lang="en"
    >
      <tbody>
        <tr height="16" style="height:16px">
          <td></td>
        </tr>
        <tr align="center">
          <td>
            <div>
              <div></div>
            </div>
            <table
              border="0"
              cellspacing="0"
              cellpadding="0"
              style="padding-bottom:20px;max-width:516px;min-width:220px"
            >
              <tbody>
                <tr>
                  <td width="8" style="width:8px"></td>
                  <td style="width: 100%;">
                    <div
                      style="border-style:solid;border-width:thin;border-color:#dadce0;padding:40px 20px;background: white;border-radius: 3px"
                      align="center"
                    >
                      <div
                        style="width: 25px;
                                          border: thin;
                                          border-color: grey;
                                          background: #ecf2fd;
                                          border-radius: 50%;
                                          padding: 25px;
                                          height: 25px;"
                      >
                        <img
                          style="display: inline-block;
                                          font-size: 2em;
                                          height: 1em;
                                          overflow: visible;
                                          vertical-align: -0.125em;
                                          width: 0.75em;
                                          color:#467fcf;"
                          src="https://d3s2030oirvd6v.cloudfront.net/ops-master-assets/unlock.png"
                        />
                      </div>
                      <div
                        style="font-family:'Source Sans Pro', -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Helvetica Neue', Arial, sans-serif;color: #454c52 !important;line-height:32px;padding-top: 20px;text-align:center;word-break:break-word"
                      >
                        <div style="font-size:24px;font-weight: 200;">
                          Password reset request
                        </div>
                      </div>
                      <div
                        style="font-family:Roboto-Regular,Helvetica,Arial,sans-serif;font-size:14px;color: #454c52 !important;line-height:20px;padding-top:20px;text-align:center"
                      >
                        <p>
                          You have requested a password reset for your GFS Kitchen Screens account.
                          <br />
                          Click the button below to change your password:
                        </p>
                        <div style="padding-top:32px;text-align:center">
                          <a
                            href="${url}"
                            style="font-family:'Source Sans Pro', -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Helvetica Neue', Arial, sans-serif;line-height:16px;color:#ffffff;font-weight:400;text-decoration:none;font-size:14px;display:inline-block;padding:10px 24px;background-color:#467fcf;border-radius:3px;min-width:90px"
                            target="_blank"
                          >
                            Reset password</a
                          >
                        </div>
                      </div>
                      <div style="text-align:left">
                        <div
                          style="font-family:Roboto-Regular,Helvetica,Arial,sans-serif;color:rgba(0,0,0,0.54);font-size:11px;line-height:18px;padding-top:50px;text-align:center"
                        >
                          <div>You can also click here:</div>
                          <a href="${url}"
                            >${url}</a
                          >
                          <div style="direction:ltr">
                            <a
                              style="font-family:Roboto-Regular,Helvetica,Arial,sans-serif;color:rgba(0,0,0,0.54);font-size:11px;line-height:18px;padding-top:12px;text-align:center"
                            >
                              You can ignore this email if you didn't request for
                              your password to be reset.</a
                            >
                          </div>
                        </div>
                      </div>
                    </div>
                  </td>
                  <td width="8" style="width:8px"></td>
                </tr>
              </tbody>
            </table>
          </td>
        </tr>
        <tr height="32" style="height:32px">
          <td></td>
        </tr>
      </tbody>
    </table>
  </div>`;
  },
};
