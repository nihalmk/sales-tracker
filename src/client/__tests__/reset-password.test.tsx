import 'jsdom-global/register';
import React from 'react';
import { shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Reset } from '../pages/reset-password/[token]';
import cookie from 'js-cookie';

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: () => ({
    query: {
      token: 'randomtoken',
    },
    push: () => {},
  }),
}));

jest.mock('../../accounts/client', () => ({
  accountsPassword: {
    resetPassword: jest.fn(() => ({
      tokens: { accessToken: 'abcd' },
    })),
  },
}));

describe('The reset password page', () => {
  let wrapper: any, passwordTxt: any, confirmPasswordTxt: any;
  beforeEach(() => {
    wrapper = shallow(<Reset />);
    passwordTxt = wrapper.find('#password').props();
    confirmPasswordTxt = wrapper.find('#confirmPassword').props();
  });

  it('renders as expected', async () => {
    const snapshot = shallow(<Reset />);
    expect(snapshot).toMatchSnapshot();
  });

  it('does not allow submit if password is blank', () => {
    const submitBtn = wrapper.find('button').first();
    expect(submitBtn.html()).toContain('disabled');
  });

  it('does not allow submit if password and confirm password are different', async () => {
    act(() => {
      passwordTxt.onChange({
        target: {
          value: 'abc',
        },
      } as React.ChangeEvent<HTMLInputElement>);
    });
    wrapper.update();
    act(() => {
      confirmPasswordTxt.onChange({
        target: {
          value: 'xyz',
        },
      } as React.ChangeEvent<HTMLInputElement>);
    });
    wrapper.update();
    const submitBtn = wrapper.find('button').first();
    expect(submitBtn.html()).toContain('disabled');
  });

  it('allows submit if password and confirm password are the same', () => {
    act(() => {
      passwordTxt.onChange({
        target: {
          value: 'abc',
        },
      } as React.ChangeEvent<HTMLInputElement>);
    });
    wrapper.update();
    act(() => {
      confirmPasswordTxt.onChange({
        target: {
          value: 'abc',
        },
      } as React.ChangeEvent<HTMLInputElement>);
    });
    wrapper.update();
    const submitButton = wrapper.find('button').first();
    expect(submitButton.html()).not.toContain('disabled');
  });

  it('sets token in cookie after a successful password reset', async () => {
    const cookieSet = jest.spyOn(cookie, 'set');
    act(() => {
      passwordTxt.onChange({
        target: {
          value: 'abc',
        },
      } as React.ChangeEvent<HTMLInputElement>);
    });
    wrapper.update();
    act(() => {
      confirmPasswordTxt.onChange({
        target: {
          value: 'abc',
        },
      } as React.ChangeEvent<HTMLInputElement>);
    });
    wrapper.update();
    const submitButton = wrapper.find('button').first();
    await act(async () => {
      submitButton.simulate('click');
    });
    expect(cookieSet).toHaveBeenCalledWith('token', 'abcd', { expires: 1 });
  });
});
