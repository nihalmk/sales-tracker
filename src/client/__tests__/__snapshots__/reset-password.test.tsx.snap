// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The reset password page renders as expected 1`] = `
<div
  className="col col-login mx-auto"
>
  <div
    className="text-center mb-6"
  >
    <Logo
      setColor={true}
    />
  </div>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <div
        className="card-title text-center"
      >
        Reset Password
      </div>
    </div>
    <div
      className="card-body p-6"
    >
      <p>
        Enter your new password below to reset the password of your account.
      </p>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          New Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Confirm New Password
        </label>
        <input
          className="form-control"
          id="confirmPassword"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="form-footer"
      >
        <button
          className="btn btn-primary btn-block "
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Reset password
        </button>
      </div>
    </div>
  </div>
</div>
`;
